packages/admin-panel-server/src/routes/dashboardVisualisations/ImportDashboardVisualisationRoute.ts

```diff
@@ -59,11 +60,40 @@ export class ImportDashboardVisualisationRoute extends Route<ImportDashboardVisu
   }
 
   public async buildResponse() {
-    if (!this.req.file) {
+    const { files } = this.req;
+    if (!files || !Array.isArray(files)) {
       throw new UploadError();
     }
 
-    const { dashboards = [], dashboardRelations = [], ...visualisation } = this.readFileContents();
+    const errors: [string, string][] = [];
+    const importedVizIds: string[] = [];
+    for (let i = 0; i < files.length; i++) {
+      const file = files[i];
+      const { originalname: fileName } = file;
+      try {
+        const fileContent = importFileSchema.validateSync(readFileContent(file));
+        importedVizIds.push(await this.importViz(fileContent));
+      } catch (error: any) {
+        errors.push([fileName, error.message]);
+      }
+    }
+
+    if (errors.length > 0) {
+      throw new UploadError(
+        new Error(errors.map(([fileName, errorMsg]) => `\n  ${fileName}: ${errorMsg}`).join('')),
+      );
```

Comment:
It is up to you that we can tell another visual is imported successfully in here.

Reply:
Ah good call!  I'll fix that